smart-contract/assembly/contracts/deweb-interface.ts

```diff
@@ -136,3 +145,51 @@ export function withdraw(binaryArgs: StaticArray<u8>): void {
 
   transferCoins(Context.caller(), amount);
 }
+
+/**
+ * Deletes a file from the contract storage.
+ * @param _binaryArgs - Serialized arguments containing the ChunkDelete object.
+ * @param index - (optional) The index of the file to delete of deleting multiple files.
+ * @throws If the file does not exist or if the caller is not the owner.
+ */
+export function deleteFile(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+  const args = new Args(_binaryArgs);
+
+  const file = args
+    .nextSerializableObjectArray<ChunkDelete>()
+    .expect('Invalid files');
+
+  assert(_isPathFileInList(file[0].filePath), 'File does not exist');
+
+  _deleteFile(file[0].filePathHash);
+
+  _removeFilePath(file[0].filePath);
+}
+
+/**
+ * Deletes a set of files from the contract storage. Calls deleteFile for each file.
+ * @param _binaryArgs - Serialized arguments containing the ChunkDelete object.
+ * @throws If the file does not exist or if the caller is not the owner.
+ */
+export function deleteFiles(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+  const args = new Args(_binaryArgs);
+  const files = args
+    .nextSerializableObjectArray<ChunkDelete>()
+    .expect('Invalid files');
+
+  for (let i = 0; i < files.length; i++) {
+    assert(_isPathFileInList(files[i].filePath), 'File does not exist');
```

Comment:
here maybe we can use 
```
  assert(_getTotalChunk(file.filePathHash) > 0, 'File does not exist');
```

It will be less gas consuming

Reply:
it depends what we consider the truth:  absence of filePath or absence of Chunks.... but yes I agree on the change